frontend/src/helpers/index.ts

```diff
@@ -71,3 +72,69 @@ export function addQuery(
     search: searchParams.toString(),
   });
 }
+
+/**
+ * @returns true if the app is in development mode.
+ */
+function isDevMode(): boolean {
+  return !process.env.NODE_ENV || process.env.NODE_ENV === 'development';
+}
+
+/**
+ * @returns URL depending on dev-mode/electron, base-url, and window.location.origin.
+ *
+ * @example isDevMode | isElectron returns 'http://localhost:4466/'
+ * @example base-url set as '/headlamp' returns '/headlamp/'
+ * @example isDevMode | isElectron and base-url is set
+ *          it returns 'http://localhost:4466/headlamp/'
+ * @example returns 'https://headlamp.example.com/'using the window.location.origin of browser
+ *
+ */
+function getAppUrl(): string {
+  let url =
+    exportFunctions.isDevMode() || exportFunctions.isElectron()
+      ? 'http://localhost:4466'
+      : window.location.origin;
+
+  const baseUrl = exportFunctions.getBaseUrl();
+  url += baseUrl ? baseUrl + '/' : '/';
+
+  return url;
+}
+
+declare global {
+  interface Window {
+    headlampBaseUrl?: string;
+  }
+}
```

Comment:
Isn't this going to clash with the same Window interface we have declared in the src/plugin/index.tsx?

Reply:
This just extends the type, so it's fine to extend the type multiple times (AFAIK).

I'm testing the plugins still work with this branch, and will leave this item open until that's done.

Reply:
Plugins work now.